src/PostProcess/RenderPipeline/babylon.postProcessRenderEffect.ts

```diff
@@ -1,34 +1,46 @@
 module BABYLON {
+    /**
+     * This represents a set of one or more post processes in Babylon.
+     * A post process can be used to apply a shader to a texture after it is rendered.
+     * @example https://doc.babylonjs.com/how_to/how_to_use_postprocessrenderpipeline
+     */
     export class PostProcessRenderEffect {
         private _postProcesses: any;
-        private _getPostProcess: () => Nullable<PostProcess | Array<PostProcess>>;
+        private _getPostProcesses: () => Nullable<PostProcess | Array<PostProcess>>;
 
         private _singleInstance: boolean;
 
         private _cameras: { [key: string]: Nullable<Camera> };
         private _indicesForCamera: { [key: string]: number[] };
 
-        private _renderEffectAsPasses: any;
-
-        // private
+        /**
+         * Name of the effect
+         */
         public _name: string;
-
-        public applyParameters: (postProcess: PostProcess) => void;
         
-        constructor(engine: Engine, name: string, getPostProcess: () => Nullable<PostProcess | Array<PostProcess>>, singleInstance?: boolean) {
+        /**
+         * Instantiates a post process render effect.
+         * A post process can be used to apply a shader to a texture after it is rendered.
+         * @param engine The engine the effect is tied to
+         * @param name The name of the effect
+         * @param getPostProcesses A function that returns a set of post processes which the effect will run in order to be run.
+         * @param singleInstance False if this post process can be run on multiple cameras. (default: true)
+         */
+        constructor(engine: Engine, name: string, getPostProcesses: () => Nullable<PostProcess | Array<PostProcess>>, singleInstance?: boolean) {
             this._name = name;
             this._singleInstance = singleInstance || true;
 
-            this._getPostProcess = getPostProcess;
+            this._getPostProcesses = getPostProcesses;

```

Comment:
Do we need that?
 single array should be enough?

Reply:
Previously _postProcesses was an array because if you apply a PP to multiple cameras names 1,2 and 3 _postProcesses = {1:pp, 2:pp, 3:pp} now because I had to extend it to support multiple post processes in the effect it will be _postProcesses = {1:[pp1, pp2,...], 2:[pp1, pp2,...], 3:[pp1, pp2,...]}

Reply:
makes sense